src/subcommands/common.rs

```diff
@@ -15,6 +19,23 @@ pub struct ProviderOpts {
     pub provider: u8,
 }
 
+impl ProviderOpts {
+    /// Get the ProviderID selected by the user or the service default if
+    /// no provider was selected.
+    ///
+    /// The Core Provider cannot be used and will be overriden by the
+    /// service default.
+    pub fn provider(&self) -> Result<ProviderID, ParsecToolError> {
+        if self.provider != 0 {
+            Ok(ProviderID::try_from(self.provider)?)
+        } else {
+            let mut client = BasicClient::new_naked();
+            client.set_default_provider()?;
+            Ok(client.implicit_provider())
```

Comment:
It's perfectly fine for now, it would be nice to add in the client static methods to return what is the current default provider/authenticator as well!

Reply:
Yep, it's exactly what I was thinking. I just couldn't be bothered to change the client for this, but maybe I should. You'd need to at least create an operation client in that "static" method anyway, so it's not that much overhead to have the whole `BasicClient`

Reply:
Yes, it's fine for now!